src/YoloDev.Expecto.TestSdk/settings.fs

```diff
@@ -14,58 +18,159 @@ type RunSettings =
     disableParallelization: bool
 
     /// Gets a value which indicates the target framework the tests are being run in.
-    targetFrameworkVersion: string option }
+    targetFrameworkVersion: string option 
+
+    /// Gets the [ExpectoConfig](https://github.com/haf/expecto#the-config) that was set via RunSettings
+    expectoConfig : ExpectoConfig}
 
 [<RequireQualifiedAccess>]
 module RunSettings = 
   let defaultSettings = 
     { collectSourceInformation = true
       designMode = true
       disableParallelization = false
-      targetFrameworkVersion = None }
-  
-  let read (runSettings: IRunSettings) = 
-    let settings = defaultSettings
+      targetFrameworkVersion = None
+      expectoConfig = ExpectoConfig.defaultConfig }
+
+  let private foldCLIArgumentToConfig = function
+    | Sequenced -> fun o -> { o with ``parallel`` = false }
+    | Parallel -> fun o -> { o with ``parallel`` = true }
+    | Parallel_Workers n -> fun o -> { o with parallelWorkers = n }
+    | Stress_Timeout n -> fun o -> { o with stressTimeout = TimeSpan.FromMinutes n }
+    | Stress_Memory_Limit n -> fun o -> { o with stressMemoryLimit = n }
+    | Fail_On_Focused_Tests -> fun o -> { o with failOnFocusedTests = true }
+    | Debug -> fun o -> { o with verbosity = Expecto.Logging.LogLevel.Debug }
+    | Log_Name name -> fun o -> { o with logName = Some name }
+    | Filter hiera -> fun o -> {o with filter = Test.filter (fun s -> s.StartsWith hiera )}
+    | Run tests -> fun o -> {o with filter = Test.filter (fun s -> tests |> List.exists ((=) s) )}
+    | FsCheck_Max_Tests n -> fun o -> {o with fsCheckMaxTests = n }
+    | FsCheck_Start_Size n -> fun o -> {o with fsCheckStartSize = n }
+    | FsCheck_End_Size n -> fun o -> {o with fsCheckEndSize = Some n }
+    | Allow_Duplicate_Names -> fun o -> { o with allowDuplicateNames = true }
+    | No_Spinner -> fun o -> { o with noSpinner = true }
+    | _ -> id
+
+  let readValueParse parser elementName confNode =
+      confNode
+      |> Option.bind (Xml.element elementName)
+      |> Option.bind Xml.value
+      |> Option.bind parser
+
+  let readValueString elementName confNode = readValueParse Option.ofObj elementName confNode
+  let readValueBool elementName confNode = readValueParse TryParse.bool elementName confNode
+  let readValueInt32 elementName confNode = readValueParse TryParse.int32 elementName confNode
+  let readValueFloat32 elementName confNode = readValueParse TryParse.float32 elementName confNode
+
+  let readExpectoConfig expectoConfig (confNode: Xml.Linq.XElement option) =
```

Comment:
Can we do this the other way around? Instead of `tryGetNamed "name" xml`, do `tryParse node`.

```fsharp
let (|Element|_|) name node =
  // guesswork on API names
  if name.Equals (node.Name, StringComparison.OrdinalIgnoreCase)
  then Some node.Value
  else None

let (|Bool|Int|Float|String|) str =
  ........

match node with
| Element "parallel" (Bool true) -> Some CLIArguments.Parallel
| Element "parallel" (Bool false) -> Some CLIArguments.Sequenced
....
```